src/pages/index.js

```diff
@@ -65,3 +65,57 @@ profileFormValidator.enableValidation();
 
 const cardsFormValidator = new FormValidator(config, cardsEditForm);
 cardsFormValidator.enableValidation();
+
+//
+
+export const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f01264e8-2101-4ab7-b120-b09b55e63681",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .returnUserInfoAndCards()
+  .then((res) => {
+    profileEdited.setUserInfo(res[0]);
+
+    propic.src = res[0].avatar;
+
+    const newCardsSection = new Section(
+      {
+        data: res[1],
+        renderer: (item) => {
+          const cardElement = createCard(item);
+          cardList.append(cardElement);
+        },
+      },
+      cardList
+    );
+    newCardsSection.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+const propicPopup = new PopupWithForm(propicModal, {
+  handleFormSubmit: (obj) => {
+    api.updateProPic(obj.link);
+    propic.src = obj.link;
+  },
+});
+propicPopup.setEventListeners();
+
+propicButtonEdit.addEventListener("click", () => {
+  propicPopup.open();
+});
+
+export const deletePopup = new PopupWithForm(deleteModal, {
+  handleFormSubmit: (cardElement, objId) => {
+    api.deleteCard(objId);
+    cardElement.remove();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.